# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/loader/URLPreloader.h
# Commit: a345e05df151
# Full Hash: a345e05df151563625fe05ec63094b3dae2c9379
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-07 21:31:11
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r 288a67aed661 -r a345e05df151 js/xpconnect/loader/URLPreloader.h
--- a/js/xpconnect/loader/URLPreloader.h	Tue Jul 07 04:35:08 2020 +0000
+++ b/js/xpconnect/loader/URLPreloader.h	Tue Jul 07 04:36:27 2020 +0000
@@ -26,8 +26,6 @@
 #include "nsIThread.h"
 #include "nsReadableUtils.h"
 
-class nsZipArchive;
-
 namespace mozilla {
 namespace loader {
 class InputBuffer;
@@ -77,7 +75,7 @@
   static Result<const nsCString, nsresult> ReadFile(nsIFile* file,
                                                     ReadType readType = Forget);
 
-  static Result<const nsCString, nsresult> ReadZip(nsZipArchive* archive,
+  static Result<const nsCString, nsresult> ReadZip(CacheAwareZipReader* archive,
                                                    const nsACString& path,
                                                    ReadType readType = Forget);
 
@@ -198,7 +196,7 @@
       return "";
     }
 
-    already_AddRefed<nsZipArchive> Archive() {
+    already_AddRefed<CacheAwareZipReader> Archive() {
       return Omnijar::GetReader(OmnijarType());
     }
 